src/main/kotlin/eu/europa/ec/eudi/pidissuer/adapter/out/jose/ValidateJwtProof.kt

```diff
@@ -94,7 +92,10 @@ private fun algorithmAndCredentialKey(
     val x5c = header.x509CertChain
 
     val key = when {
-        kid != null && jwk == null && x5c.isNullOrEmpty() -> CredentialKey.DIDUrl(kid)
+        kid != null && jwk == null && x5c.isNullOrEmpty() -> {
+            val url = DIDURL.fromString(kid)
```

Comment:
Perhaps these two lines can be moved to the companion object of `CredentialKey.DIDUrl` as a factory method which assures that we are dealing only with DIDs that we can resolve.